external/model-preparation-algorithm/mpa_tasks/apis/detection/task.py

```diff
@@ -138,9 +138,9 @@ def _infer_detector(
         output = results["outputs"]
         metric = output["metric"]
         predictions = output["detections"]
-        assert len(output["detections"]) == len(output["feature_vectors"]) == len(output["saliency_maps"]), (
+        assert len(output["detections"]) == len(output["feature_vectors"]), (
             "Number of elements should be the same, however, number of outputs are "
-            f"{len(output['detections'])}, {len(output['feature_vectors'])}, and {len(output['saliency_maps'])}"
+            f"{len(output['detections'])}, {len(output['feature_vectors'])})"
         )
```

Comment:
Does it cause test failure?

Yes, total number of saliency maps is `len(dataset) * N` where N is the number of classes. But single saliency map is actually a 3-dimentional tensor [N, H, W], where N is the number of classes. So basically `len(output["feature_vectors"]) == len(output["saliency_maps"])` actually should be `True`.